src/pair_mie_cut.f90

```diff
@@ -0,0 +1,131 @@
+!    (at your option) any later version.
+!
+!    EmDee is distributed in the hope that it will be useful,
+!    but WITHOUT ANY WARRANTY; without even the implied warranty of
+!    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+!    GNU General Public License for more details.
+!
+!    You should have received a copy of the GNU General Public License
+!    along with EmDee. If not, see <http://www.gnu.org/licenses/>.
+!
+!    Author: Charlles R. A. Abreu (abreu@eq.ufrj.br)
+!            Applied Thermodynamics and Molecular Simulation
+!            Federal University of Rio de Janeiro, Brazil
+
+module pair_mie_cut_module
+
+use global
+use pairModelClass
+
+implicit none
+
+!> Abstract class for pair model mie
+!!
+!! NOTES: 1) model parameters must be declared individually and tagged with a comment mark "!<>"
+!!        2) recognizable parameter types are real(rb) and integer(ib)
+!!        3) allocatable one-dimensional arrays (i.e. vectors) are permitted as parameters
+!!        4) an integer(ib) scalar parameter - a size - must necessarily succeed every allocatable
+!!           parameter or series of equally-sized allocatable parameters.
+
+type, extends(cPairModel) :: pair_mie_cut
+  real(rb) :: epsilon !<> Depth of the potential well
+  real(rb) :: sigma   !<> Distance at which the potential is zero
+  real(rb) :: n !<> Softness/hardness of the repulsion
+  real(rb) :: m !<>  Range of the attraction of the intermolecular potential
+  real(rb) :: miem,mien
+  contains
+    procedure :: setup => pair_mie_cut_setup
+    procedure :: compute => pair_mie_cut_compute
+    procedure :: virial => pair_mie_cut_virial
+    procedure :: mix => pair_mie_cut_mix
+end type pair_mie_cut
+
+contains
+
+!---------------------------------------------------------------------------------------------------
+
+  subroutine pair_mie_cut_setup( model, params, iparams )
+    class(pair_mie_cut), intent(inout) :: model
+    real(rb), intent(in), optional :: params(:)
+    integer,  intent(in), optional :: iparams(:)
+    real(rb) :: coeff
+    ! Model name:
+    model%name = "mie_cut"
+
+    ! Model parameters:
+    model%epsilon = params(1)
+    model%sigma = params(2)
+    model%n = params(3)
+    model%m = params (4)
+    ! Pre-computed quantities:
+    coeff  = (model%n / (model%n - model%m)) * (model%n / model%m) ** (model%m / (model%n - model%m)) * model%epsilon
+    model%mien = coeff * model%sigma ** model%n
+    model%miem = coeff * model%sigma ** model%m
+
+  end subroutine pair_mie_cut_setup
+
+!---------------------------------------------------------------------------------------------------
+
+  subroutine pair_mie_cut_compute( model, Eij, Wij, invR, invR2 )
+    class(pair_mie_cut), intent(in)  :: model
+    real(rb),           intent(out) :: Eij, Wij, invR
+    real(rb),           intent(in)  :: invR2
+
+    real(rb) :: rn, rm
+
+    rm = invr2 ** (model%m / 2.0_rb)
+    rn = rm ** (model%n / model%m)
```

Comment:
An exponentiation a^n with real exponent is a very slow operation. Supposing that it is as slow as doing exp(n*log(a)), we could save time here by rewriting the code as something like:
logInvR2 = log(invR2)
rm = exp(halfm*logInvR2)
rn = exp(halfn*logInvR2)
This would involve two exp's, but only one log. I think we should test the two alternatives and select the fastest one.